### PR TITLE
Update functionality with FinanceDatabase v2, depreciate `stocks/sia`, fix `exe --example` and fix `stocks/fa/dcf`

### LINKED ISSUE #4225: [Bug] Stocks/DCF - Function is broken
This is almost certainly yFinance-related, which begs the question of why it needs to even go there? The source is "StockAnalysis", and helper functions should not be calling yFinance to get metadata. 

```
2023 Feb 12, 16:29 (🦋) /stocks/fa/ $ dcf

Error: Could not generate a dataframe for the ticker symbol

Error: Could not generate a dataframe for the ticker symbol

Error: Could not generate a dataframe for the ticker symbol

WARNING: No decryption keys could be extracted from JS file. Falling back to backup decrypt methods.
Error: 'list' object has no attribute 'shape'

Error: not enough values to unpack (expected 3, got 0)
```

### PR DESCRIPTION
**Pull request by JerBouma**
# Description

Basically what changed in this new version of the FinanceDatabase is the following:

- Package itself is object orientated, needs initialisation (collect data first) but gets the job done in a much quicker fashion. For example a file with 180k tickers is loaded within 2 seconds and from there on everything is done locally and really quick. 
- The Database itself is now CSV files meaning it is much easier to edit the files. No longer thousands of JSON files that have the same data multiple times.
- All files loaded into the package are compressed down versions of Pickles with compression method `xz`. It shrinks the original JSON file of 120mb to 10mb, see [this file](https://github.com/JerBouma/FinanceDatabase/blob/main/compression/compression.ipynb). Auto-updated with GitHub Actions.
- Querying is easier with `.search` using kwargs.
- Equities now follows GICS sectors, industry groups and industries.
- ETFs and Funds have their categories reduced and grouped. Furthermore, families (e.g. Vanguard) now is grouped better (there were 5 different versions of their name).
- Multiple corrections within the entire database.
- Revamp of the whole thing to entice people to help out and update (also given that the CSVs now exist). Whether this actually happens is of course yet to be seen.

For OpenBB Terminal this means:
- `stocks/search`is faster.
- `etf` search functions are faster and updates are more frequent.
- `stocks/sia` is less buggy with less dependency on Yahoo Finance. **However, this PR also retires this menu for the time being. This is because it's currently functionality is better resolved within `stocks/ca` and through `routines`.**
- Some minor misc updates, making code more efficient and requiring far less calls to the web (specifically GitHub).
- Also fixed `stocks/fa/dcf` since it relied partly on FinanceDatabase so I extended that to replace Yahoo Finance functionality.

Fixes #4225

# How has this been tested?

* Please describe the tests that you ran to verify your changes.
* Provide instructions so we can reproduce.
* Please also list any relevant details for your test configuration.
- [x] Make sure affected commands still run in terminal
- [x] Ensure the SDK still works
- [x] Check any related reports

# Checklist:

- [x] I have adhered to the GitFlow naming convention and my branch name is in the format of `feature/feature-name` or `hotfix/hotfix-name`.
- [x] Update [our documentation](https://openbb-finance.github.io/OpenBBTerminal/) following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/website).  Update any user guides that are affected by the changes.
- [x] Update our tests following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/tests).
- [x] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/blob/main/CONTRIBUTING.md).
- [x] If a feature was added make sure to add it to the corresponding [integration test script](https://github.com/OpenBB-finance/OpenBBTerminal/tree/develop/openbb_terminal/miscellaneous/integration_tests_scripts).

# Others
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.

### CODE REVIEW

**Comment by JerBouma** [comment]
Done, not entirely sure what's with the infinity mark regarding files 😅

**Comment by hjoaquim** [comment]
@JerBouma did you use poetry to update `financedatabase` dependency?

**Comment by JerBouma** [comment]
@hjoaquim I didn't, it failed on me when I did that 😅

**Comment by hjoaquim** [comment]
Sorry to comment like this but, can't comment directly on VSCode:

- Is this function being used anywhere or we can deprecate it `openbb_terminal.stocks.discovery.financedatabase_view.show_equities`?
- On `dcf_view` is there a way to remove this `yf.Ticker(symbol).fast_info` (similarly to what was done for the info)?

**Comment by JerBouma** [comment]
> Sorry to comment like this but, can't comment directly on VSCode:
> 
> * Is this function being used anywhere or we can deprecate it `openbb_terminal.stocks.discovery.financedatabase_view.show_equities`?
> * On `dcf_view` is there a way to remove this `yf.Ticker(symbol).fast_info` (similarly to what was done for the info)?

First one is indeed not used. I just updated it for the sake of it. Within the `dcf_view` I actually believe that already has no value to begin with.

**Comment by JerBouma** [comment]
Just a headsup, this PR also deletes `sia` from the OpenBB SDK. Given that we do not maintain it anymore within the OpenBB Terminal for the time being it wouldn't make sense to keep it around in the OpenBB SDK. Furthermore, it also gets rid of the documentation pages related to it.

**Comment by JerBouma** [comment]
Fixed the fact that it didn't actually search in short_name and the index properly. It does now with [7375b93](https://github.com/OpenBB-finance/OpenBBTerminal/pull/4319/commits/7375b939f96c4f85231c189bc67cd69da6acf486) so you can do the following:

```
(🦋) /stocks/ $ search --query tsla --all-exchanges

                                                      Companies found on term tsla                                                      
┏━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━┓
┃           ┃ Name                        ┃ Country       ┃ Sector                 ┃ Industry Group           ┃ Industry    ┃ Exchange ┃
┡━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━┩
│ TSLA.BA   │ Tesla, Inc.                 │ United States │ Consumer Discretionary │ Automobiles & Components │ Automobiles │ BUE      │
├───────────┼─────────────────────────────┼───────────────┼────────────────────────┼──────────────────────────┼─────────────┼──────────┤
│ TSLA.MI   │ Tesla, Inc.                 │ United States │ Consumer Discretionary │ Automobiles & Components │ Automobiles │ MIL      │
├───────────┼─────────────────────────────┼───────────────┼────────────────────────┼──────────────────────────┼─────────────┼──────────┤
│ TSLA.MX   │ Tesla, Inc.                 │ United States │ Consumer Discretionary │ Automobiles & Components │ Automobiles │ MEX      │
├───────────┼─────────────────────────────┼───────────────┼────────────────────────┼──────────────────────────┼─────────────┼──────────┤
│ TSLA      │ Tesla, Inc.                 │ United States │ Consumer Discretionary │ Automobiles & Components │ Automobiles │ NMS      │
├───────────┼─────────────────────────────┼───────────────┼────────────────────────┼──────────────────────────┼─────────────┼──────────┤
│ TSLA.VI   │ Tesla, Inc.                 │ United States │ Consumer Discretionary │ Automobiles & Components │ Automobiles │ VIE      │
├───────────┼─────────────────────────────┼───────────────┼────────────────────────┼──────────────────────────┼─────────────┼──────────┤
│ TSLA34.SA │ Tesla, Inc.                 │ United States │ Consumer Discretionary │ Automobiles & Components │ Automobiles │ SAO      │
├───────────┼─────────────────────────────┼───────────────┼────────────────────────┼──────────────────────────┼─────────────┼──────────┤
│ JS3HN9.TI │ GSF BIDU/NIO/TSLA CC 310325 │ nan           │ nan                    │ nan                      │ nan         │ TLO      │
├───────────┼─────────────────────────────┼───────────────┼────────────────────────┼──────────────────────────┼─────────────┼──────────┤
│ TSLA-U.TI │ TESLA                       │ nan           │ nan                    │ nan                      │ nan         │ TLO      │
├───────────┼─────────────────────────────┼───────────────┼────────────────────────┼──────────────────────────┼─────────────┼──────────┤
│ TSLAC.BA  │ TESLA INC                   │ nan           │ nan                    │ nan                      │ nan         │ BUE      │
├───────────┼─────────────────────────────┼───────────────┼────────────────────────┼──────────────────────────┼─────────────┼──────────┤
│ TSLAD.BA  │ TESLA INC                   │ nan           │ nan                    │ nan                      │ nan         │ BUE      │
└───────────┴─────────────────────────────┴───────────────┴────────────────────────┴──────────────────────────┴─────────────┴──────────┘
```

**Comment by JerBouma** [comment]
I've combined `short_name` and `long_name` in the files therefore please wait before merging since otherwise it will break on next update.

**Comment by jmaslek** [comment]
> I've combined `short_name` and `long_name` in the files therefore please wait before merging since otherwise it will break on next update.

Waiting until when?

**Comment by JerBouma** [comment]
This weekend, can also merge and I hotfix this weekend as it is a small little change.

**Comment by jmaslek** [comment]
> This weekend, can also merge and I hotfix this weekend as it is a small little change.

infinity + 1 = infinity

**Comment by JerBouma** [comment]
Fixed everything, should now be ready for review. The force push was because of this:

image

**Comment by JerBouma** [comment]
@hjoaquim Can you bump `financedatabase` to 2.0.6, I keep trying to make it work but poetry fails on me. When I do add or install, it basically just deletes all the packages.

**Comment by JerBouma** [comment]
Small fix here so that it doesn't crash if you have no ticker loaded.

image

@jmaslek Let me know if I need to give you a tour what I changed due to `infinite` files.

**Comment by jmaslek** [comment]
> Small fix here so that it doesn't crash if you have no ticker loaded.
> 
> image
> 
> @jmaslek Let me know if I need to give you a tour what I changed due to `infinite` files.

found what infinite means to github:  
Screenshot 2023-02-25 at 6 13 54 PM

**Comment by jmaslek** [comment]
etf/screener_controller can be deleted (with associated tests)

**Comment by jmaslek** [comment]
On stocks/load I am not getting autocomplete options with nay of the choices :(

**Comment by jmaslek** [comment]
> On stocks/load I am not getting autocomplete options with nay of the choices :(

Related to this:  when I go into stocks I get this message.

```
2023 Feb 25, 18:31 (🦋) / $ stocks

Note: Some datasets from GitHub failed to load. This means that the `search` command and the /stocks/sia menu will not work. If other commands are failing please check your internet connection or communicate with your IT department that certain websites are blocked.
```

Am at home.  Have always done this from home.  (Probably also want to get rid of the stocks/sia reference there)

**Comment by JerBouma** [comment]
You get that because you are not on the latest version of the package, v2.0.6. Thats also why no auto complete in `search`.

**Comment by jmaslek** [comment]
> You get that because you are not on the latest version of the package, v2.0.6. Thats also why no auto complete in `search`.

Hm.  Alright. Poetry install is giving 2.0.4 so we need to bump that.

**Comment by jmaslek** [comment]
> > You get that because you are not on the latest version of the package, v2.0.6. Thats also why no auto complete in `search`.
> 
> Hm. Alright. Poetry install is giving 2.0.4 so we need to bump that.

fixed

**Comment by JerBouma** [comment]
@jmaslek 

Updated package to v2.0.7 (saw minor mistake in my own code) and removed `etf/screener`. Anything else before merge?

**Comment by jmaslek** [comment]
> @jmaslek
> 
> Updated package to v2.0.7 (saw minor mistake in my own code) and removed `etf/screener`. Anything else before merge?

Just make sure tests are passing

File: pyproject.toml [hunk 1]
```diff
@@ -52,7 +52,7 @@ bt = "^0.2.9"
 vaderSentiment = "3.3.2"
 GitPython = "^3.1.17"
 fred = "^3.1"
-financedatabase = "^2.0.6"
+financedatabase = {version = "2.0.7", extras = ["all"]}
```
Comment (jmaslek) [inline pyproject.toml:55]: this wont install on poetry install.  To do this, you want to run `poetry add financedatatbase@latest`

Then 
```
poetry export -f requirements.txt --without-hashes --output requirements.txt
poetry export -f requirements.txt --with dev -E installer -E all --without-hashes --output requirements-full.txt
```

**Review by jmaslek — APPROVED** [review]
small changes.  lgtm

**Pull request merged**